wordpress-mobile/WordPress-Android
# Dashboard Personalization: Recent Activity Log Card

Parent #18948 

This is the first step in a series of changes aimed at adding a "more menu" to the recent Activity Log dashboard card. 

Menu Closed | Menu Open
--|--
<img width="250" height="500" alt="Alt desc" src="https://github.com/wordpress-mobile/WordPress-Android/assets/506707/f3ae97a5-8eab-4e85-80a8-d90f00de9023"> | <img width="250" height="500" alt="Alt desc" src="https://github.com/wordpress-mobile/WordPress-Android/assets/506707/def50b01-6ae4-4e56-bf01-486ec22c9f40">

**Included in this PR are the following:**
- Added "more menu" to the recent activity log dashboard card
- Handle More Menu item clicks
- Removed the "footer link" portion of the card
- Removed tracking related to footer link
- Removed navigation events related to footer link
- Updated unit and instrumentation tests 

**Not included is this PR**: 
- Tracking events for the menu item taps 
- Logic to hide the card when "hide this" is tapped

**Note**: No dashboard cards retain open menus upon rotation 

**To test:**
- Install and login to the app
- Select a site in which you have access to Activity Log
- Tap My Site
- Scroll to the Recent Activity card
- ✅ Verify the card does not contain a footer 
- ✅ Verify the card has a vertical more menu
- Tap on the More Menu 
- ✅ Verify the more menu is expanded
- Tap on the All activity menu item
- ✅ Verify you are navigated to the Activity Log view


## Regression Notes
1. Potential unintended areas of impact
The activity log card does not work properly

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Updated automated and unit tests

3. What automated tests I added (or what prevented me from doing so)
MySiteViewModelTest, CardsBuilderTest, CardsShownTrackerTest, ActivityCardBuilderTest

PR submission checklist:

UI Changes testing checklist:
